missing intrinsics in gcc-7.2

I get this error message when compiling with `-mavx512f -mavx512cd -mavx512bw -mavx512dq -mavx512vl`
```console
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h: In member function ‘UME::SIMD::SIMDVec_u<unsigned int, 8> UME::SIMD::SIMDVec_u<unsigned int, 8>::swizzle(const UME::SIMD::SIMDSwizzle<8>&) const’:
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h:288:26: error: ‘_mm256_permutexvar_epi32’ was not declared in this scope
             __m256i t0 = _mm256_permutexvar_epi32(mVec, sMask.mVec);
                          ^~~~~~~~~~~~~~~~~~~~~~~~
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h:288:26: note: suggested alternative: ‘_mm256_permutex2var_epi32’
             __m256i t0 = _mm256_permutexvar_epi32(mVec, sMask.mVec);
                          ^~~~~~~~~~~~~~~~~~~~~~~~
                          _mm256_permutex2var_epi32
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h: In member function ‘UME::SIMD::SIMDVec_u<unsigned int, 8> UME::SIMD::SIMDVec_u<unsigned int, 8>::swizzle()’:
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h:309:26: error: there are no arguments to ‘_mm256_permutexvar_epi32’ that depend on a template parameter, so a declaration of ‘_mm256_permutexvar_epi32’ must be available [-fpermissive]
             __m256i t1 = _mm256_permutexvar_epi32(mVec, t0);
                          ^~~~~~~~~~~~~~~~~~~~~~~~
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h:309:26: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h: In member function ‘UME::SIMD::SIMDVec_u<unsigned int, 8>& UME::SIMD::SIMDVec_u<unsigned int, 8>::swizzlea(const UME::SIMD::SIMDSwizzle<8>&)’:
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h:330:20: error: ‘_mm256_permutexvar_epi32’ was not declared in this scope
             mVec = _mm256_permutexvar_epi32(mVec, sMask.mVec);
                    ^~~~~~~~~~~~~~~~~~~~~~~~
seqan-src/include/umesimd/plugins/avx512/uint/UMESimdVecUint32_8.h:330:20: note: suggested alternative: ‘_mm256_permutex2var_epi32’
             mVec = _mm256_permutexvar_epi32(mVec, sMask.mVec);
                    ^~~~~~~~~~~~~~~~~~~~~~~~
                    _mm256_permutex2var_epi32
```

-----

The code is

https://github.com/edanor/umesimd/blob/a3f54e5c93b9b75d83e037f3dc3c0b8ee85156e0/plugins/avx512/uint/UMESimdVecUint32_8.h#L280-L298

I don't understand, why GCC <=6.4 is using `_mm256_permutevar8x32_epi32` and 7.x is using 
```c++
             __m512i t0 = _mm512_castsi256_si512(sMask.mVec); 
             __m512i t1 = _mm512_castsi256_si512(mVec); 
             __m512i t2 = _mm512_permutexvar_epi32(t0, t1); 
             __m256i t3 = _mm512_castsi512_si256(t2); 
```
Is that faster?

--- a/plugins/UMESimdPluginAVX512.h
+++ b/plugins/UMESimdPluginAVX512.h
@@ -54,7 +54,8 @@
 #define WA_GCC_INTR_SUPPORT_6_4
 #endif
 
-#if __GNUC__ < 7 || (__GNUC__ == 7 && (__GNUC_MINOR <= 1)) && ! defined(__INTEL_COMPILER)
+// WA: missing intrinsics in GCC 7.x (It is unlikely that gcc 7.x will fix those issues)
+#if __GNUC__ < 7 || (__GNUC__ == 7 && (__GNUC_MINOR <= 9)) && ! defined(__INTEL_COMPILER)
 #define WA_GCC_INTR_SUPPORT_7_1
 #endif
 

--- a/plugins/avx512/uint/UMESimdVecUint32_8.h
+++ b/plugins/avx512/uint/UMESimdVecUint32_8.h
@@ -284,7 +284,7 @@ namespace SIMD {
             __m256i t0  = _mm256_permutevar8x32_epi32(mVec, sMask.mVec);
             return SIMDVec_u(t0);
 #else
-  #if defined(__AVX512VL__)
+  #if defined(__AVX512VL__) && !defined(WA_GCC_INTR_SUPPORT_7_1)
             __m256i t0 = _mm256_permutexvar_epi32(mVec, sMask.mVec);
             return SIMDVec_u(t0);
   #else
@@ -304,7 +304,7 @@ namespace SIMD {
             __m256i t1  = _mm256_permutevar8x32_epi32(mVec, t0);
             return SIMDVec_u(t1);
 #else
-  #if defined(__AVX512VL__)
+  #if defined(__AVX512VL__) && !defined(WA_GCC_INTR_SUPPORT_7_1)
             __m256i t0 = _mm256_setr_epi32(i0, i1, i2, i3, i4, i5, i6, i7);
             __m256i t1 = _mm256_permutexvar_epi32(mVec, t0);
             return SIMDVec_u(t1);
@@ -326,7 +326,7 @@ namespace SIMD {
             mVec  = _mm256_permutevar8x32_epi32(mVec, sMask.mVec);
             return *this;
 #else
-  #if defined(__AVX512VL__)
+  #if defined(__AVX512VL__) && !defined(WA_GCC_INTR_SUPPORT_7_1)
             mVec = _mm256_permutexvar_epi32(mVec, sMask.mVec);
             return *this;
   #else

